src/createServer.js

```diff
@@ -1,9 +1,28 @@
 /* eslint-disable no-console */
 'use strict';
 
+const http = require('http');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const server = http.createServer((req, res) => {
+    const [pathname, params] = req.url.split('?');
+
+    res.writeHead(200, {
+      'Content-Type': 'application/json',
+    });
+
+    const parts = pathname.split('/').filter(Boolean);
+    const query = Object.fromEntries(new URLSearchParams(params));
```

Comment:
The use of `URLSearchParams` is correct for parsing query parameters, but ensure that `params` is not undefined. You might want to add a check or default value to handle cases where there are no query parameters.